core/common/src/main/java/alluxio/util/ConfigurationUtils.java

```diff
@@ -66,19 +68,64 @@
+   * job-master-raft).
+   *
+   * @param conf configuration
+   * @param serviceType the service to get addresses for
+   * @return the addresses
+   */
+  public static List<InetSocketAddress> getEmbeddedJournalAddresses(AlluxioConfiguration conf,
+      ServiceType serviceType) {
+    Preconditions.checkState(
+        serviceType == ServiceType.MASTER_RAFT || serviceType == ServiceType.JOB_MASTER_RAFT);
+    PropertyKey addressKey = PropertyKey.MASTER_EMBEDDED_JOURNAL_ADDRESSES;
+    if (serviceType == ServiceType.JOB_MASTER_RAFT
+        && conf.isSet(PropertyKey.JOB_MASTER_EMBEDDED_JOURNAL_ADDRESSES)) {
+      addressKey = PropertyKey.JOB_MASTER_EMBEDDED_JOURNAL_ADDRESSES;
+    }
+    List<InetSocketAddress> inetAddresses = new ArrayList<>();
+    if (conf.isSet(addressKey)) {
+      List<String> addresses = conf.getList(addressKey, ",");
+      for (String address : addresses) {
+        try {
+          InetSocketAddress addr = NetworkAddressUtils.parseInetSocketAddress(address);
+          // When the user configures the master embedded journal addresses but not the job master
+          // embedded journal addresses, we derive the job master embedded journal addresses by
+          // combining the master hostnames with the job master embedded journal port.
+          if (serviceType == ServiceType.JOB_MASTER_RAFT
+              && addressKey == PropertyKey.MASTER_EMBEDDED_JOURNAL_ADDRESSES) {
+            addr = new InetSocketAddress(addr.getHostString(),
+                conf.getInt(PropertyKey.JOB_MASTER_EMBEDDED_JOURNAL_PORT));
+          }
+          inetAddresses.add(addr);
+        } catch (IOException e) {
+          throw new IllegalArgumentException(
+              String.format("Failed to parse address %s for property %s", address, addressKey), e);
+        }
+      }
+    } else {
+      inetAddresses = Arrays.asList(NetworkAddressUtils.getConnectAddress(serviceType, conf));
+    }
+    return inetAddresses;
+  }
+
   /**
    * Gets the RPC addresses of all masters based on the configuration.
    *
    * @param conf the configuration to use
    * @return the master rpc addresses
    */
   public static List<InetSocketAddress> getMasterRpcAddresses(AlluxioConfiguration conf) {
+    // First check whether rpc addresses are explicitly configured.
     if (conf.isSet(PropertyKey.MASTER_RPC_ADDRESSES)) {
       return parseInetSocketAddresses(conf.getList(PropertyKey.MASTER_RPC_ADDRESSES, ","));
-    } else {
-      int rpcPort = NetworkAddressUtils.getPort(NetworkAddressUtils.ServiceType.MASTER_RPC, conf);
-      return getRpcAddresses(PropertyKey.MASTER_EMBEDDED_JOURNAL_ADDRESSES, rpcPort, conf);
     }
+
+    // Fall back on server-side journal configuration.
+    int rpcPort = NetworkAddressUtils.getPort(NetworkAddressUtils.ServiceType.MASTER_RPC, conf);
+    return overridePort(getEmbeddedJournalAddresses(conf, ServiceType.MASTER_RAFT), rpcPort);
```

Comment:
`ConfigurationUtils.getMasterRpcAddresses()` now only get the Master addresses when multiple masters configurations are provided (MASTER_RPC_ADDRESSES, EMBEDDED_JOURNAL_ADDRESSES), but it will be good if it can also get the master rpc addresses when only single master is provided (Only MASTER_HOSTNAME is provided). 

Basically in https://github.com/ZacBlanco/microbench/pull/1/files, first i think `ConfigurationUtils.getMasterRpcAddresses()`  will also get the single master rpc address and turns out, it did not.

Up to you to think should we include the single master address.

Reply:
I think it does work for single master. I just added a couple additional tests to show this: https://github.com/Alluxio/alluxio/pull/8516/commits/834b9d6cc211a16c70d599473fb72c8311af9259